python/ray/experimental/dask/callbacks.py

```diff
@@ -0,0 +1,197 @@
+import contextlib
+
+from dask.callbacks import Callback, unpack_callbacks
+
+
+class RayDaskCallback(Callback):
+    """
+    Extends Dask's `Callback` class with Ray-specific hooks. When instantiating
+    or subclassing this class, both the normal Dask hooks (e.g. pretask,
+    posttask, etc.) and the Ray-specific hooks can be provided.
+
+    See `dask.callbacks.Callback` for usage.
+
+    Caveats: Any Dask-Ray scheduler must bring the Ray-specific callbacks into
+    context using the `local_ray_callbacks` context manager, since the built-in
+    `local_callbacks` context manager provided by Dask isn't aware of this
+    class.
+    """
+
+    # Set of active Ray-specific callbacks.
+    ray_active = set()
+
+    def __init__(
+            self,
+            ray_presubmit=None,
+            ray_postsubmit=None,
+            ray_pretask=None,
+            ray_posttask=None,
+            ray_postsubmit_all=None,
+            ray_finish=None,
+            **kwargs,
+    ):
+        """
+        Ray-specific callbacks:
+            - def _ray_presubmit(task, key, deps):
+                Run before submitting a Ray task. If this callback returns a
```

Comment:
Could you also document the args for these callbacks? Unless these are standard for Dask callbacks (I'm not familiar with that API).